TUM Campus App/MVGManager.swift

```diff
@@ -23,22 +25,39 @@ final class MVGManager: Manager, SimpleSingleManager {
         self.config = config
     }
     
-    func fetchSingle() -> Response<DataElement?> {
-        return fetch().flatMap { stations in
-            guard let station = stations.first else { return .successful(with: nil) }
-            return self.fetch(for: station).map { departures in
-                return DetailedStation(station: station, departures: departures)
-            }
+//    func fetch() -> Response<[Station]> {
+//        return config.mvg.doObjectsRequest(to: .getNearbyStations,
+//                                           queries: ["latitude" : location.coordinate.latitude,
+//                                                     "longitude" : location.coordinate.longitude],
+//                                           at: ["locations"]).map { $0.sorted(byLocation: \.location) }
+//    }
+    
+    func fetch() -> Response<[Station]> {
+        let promise: Response<[Station]> = config.mvg.doObjectsRequest(to: .getNearbyStations,
+                                                   queries: ["latitude" : location.coordinate.latitude,
+                                                             "longitude" : location.coordinate.longitude],
+                                                   at: ["locations"]).map { $0.sorted(byLocation: \.location) }
+        return promise.flatMap { (stations: [Station]) in
+            return self.fetchDepartures(for: stations)
         }
     }
     
-    func fetch() -> Response<[Station]> {
-        return config.mvg.doObjectsRequest(to: .getNearbyStations,
-                                           queries: ["latitude" : location.coordinate.latitude,
-                                                     "longitude" : location.coordinate.longitude],
-                                           at: ["locations"]).map { $0.sorted(byLocation: \.location) }
+    func fetchDepartures(for stations: [Station]) -> Response<[Station]> {
+        
+        let map = stations.flatMap {self.fetchDeparture(for: $0)}
+        
+        let bulk = map.bulk.onSuccess { return $0 }
+        return bulk
     }
```

Comment:
@mathiasquintero I am sure this can be improved 😄

Reply:
I'll have to check it out. Can you briefly explain what was there before and what you're doing now?

Reply:
This should fetch the departures for a station, just like the mensa manager fetches the mensa menu for a mensa.